src/validators/stats/index.js

```diff
@@ -20,57 +43,44 @@ export default {key: 'stats', validate: validateStats}
 export {validKeysJoined, validStringValuesJoined, validDataTypesJoined, validateStats}
 
 
-function validateStats(val) {
-  const result = validators.reduce((res, validator) => {
-    if (res !== CONTINUE) {
-      return res
-    }
-    return validator(val)
-  }, CONTINUE)
-
-  if (result !== CONTINUE && result !== EXIT_EARLY) {
-    return result
+function invariant(condition, message) {
+  if (!condition) {
+    return message
   }
 }
 
+function validate(val, validators) {
+  const validatorWithReturnValue = _.find(validators, validator => validator(val))
+  return validatorWithReturnValue ? validatorWithReturnValue(val) : undefined
+}
+
+function validateStats(val) {
+  return validate(val, [
+    validateDataType,
+    validateAsString,
+    validateAsObject,
+  ])
+}
+
 function validateDataType(val) {
   const type = friendlyTypeOf(val)
-  const isCorrectType = includes(validDataTypes, type)
-  if (!isCorrectType) {
-    return `Unexpected type: ${type}. Valid types are: ${validDataTypesJoined}`
-  } else {
-    return CONTINUE
-  }
+  return invariant(
+    validDataTypes.includes(type),
+    `Unexpected type: ${type}. Valid types are: ${validDataTypesJoined}`
+  )
 }
 
 function validateAsString(val) {
-  if (friendlyTypeOf(val) !== 'string') {
-    return CONTINUE
-  }
-
-  const isCorrectStringValue = includes(validStringValues, val)
-  if (isCorrectStringValue) {
-    return EXIT_EARLY
```

Comment:
This was the only reason I did the reducer rather than using find, so I could exit early. I'm thinking that this isn't enough of a reason now because your code is much cleaner. Let's merge this and revert my commit to add validateReducer to configuration-validator. This is much better.